src/main/java/org/jabref/gui/maintable/PersistenceVisualStateTable.java

```diff
@@ -26,10 +26,30 @@ public PersistenceVisualStateTable(final MainTable mainTable, JabRefPreferences
         this.preferences = preferences;
 
         mainTable.getColumns().addListener(this::onColumnsChanged);
-        mainTable.getColumns().forEach(col -> col.sortTypeProperty().addListener(obs -> updateColumnSortType(col.getText(), col.getSortType())));
+        mainTable.getColumns().forEach(col -> col.sortTypeProperty().addListener(obs ->
+                updateColumnSortType(col.getText(), col.getSortType())));
+        mainTable.getColumns().forEach(col -> col.widthProperty().addListener(obs -> onColumnWidthChange()));
 
     }
 
+    private void onColumnWidthChange() {
```

Comment:
This method appears to be identical to `updateColumnPreferences` below, or do I miss something?